From 4599906069d9b3492a16e13c56c2c15a247ddfd0 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Fri, 13 Apr 2018 13:25:04 +0300
Subject: [PATCH 289/448] bus: fsl-mc: move fsl_mc_device_remove to common header

VFIO fsl-mc driver need this function on device remove,
moving this to common header file

Signed-off-by: Bharat Bhushan <Bharat.Bhushan@nxp.com>
Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
---
 drivers/bus/fsl-mc/fsl-mc-private.h |    2 --
 include/linux/fsl/mc.h              |    2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/bus/fsl-mc/fsl-mc-private.h b/drivers/bus/fsl-mc/fsl-mc-private.h
index e95c4c4..55b120a 100644
--- a/drivers/bus/fsl-mc/fsl-mc-private.h
+++ b/drivers/bus/fsl-mc/fsl-mc-private.h
@@ -550,8 +550,6 @@ int __must_check fsl_mc_device_add(struct fsl_mc_obj_desc *obj_desc,
 				   const char *driver_override,
 				   struct fsl_mc_device **new_mc_dev);
 
-void fsl_mc_device_remove(struct fsl_mc_device *mc_dev);
-
 int __init dprc_driver_init(void);
 
 void dprc_driver_exit(void);
diff --git a/include/linux/fsl/mc.h b/include/linux/fsl/mc.h
index ec526ed..ae0997a 100644
--- a/include/linux/fsl/mc.h
+++ b/include/linux/fsl/mc.h
@@ -357,6 +357,8 @@ struct fsl_mc_io {
 	module_driver(__fsl_mc_driver, fsl_mc_driver_register, \
 		      fsl_mc_driver_unregister)
 
+void fsl_mc_device_remove(struct fsl_mc_device *mc_dev);
+
 /*
  * Macro to avoid include chaining to get THIS_MODULE
  */
-- 
1.7.1

